src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/Tools/ConvertTo-AzureRmVhd.ps1

```diff
@@ -0,0 +1,160 @@
+        }
+        [string[]]$vmDiskFileNames = @();
+        foreach ($hd in $vmObj.HardDrives)
+        {
+            $vmDiskFileNames += (Split-Path -Path $hd.Path -Leaf);
+        }
+    
+        # Start the real export process
+        $st = Export-VM -ComputerName $computerName -Name $vmName -Path $exportDir;
+  
+        # Construct the final VHD file paths
+        [string[]]$vhdFiles = @();
+        foreach ($fileName in $vmDiskFileNames)
+        {
+            $vhdFullName = Join-Path (Get-HyperVVhdFolder $exportDir $vmName) $fileName;
+            if (-not (Test-Path $vhdFullName))
+            {
+                throw "The VHD file '$vhdFullName' should have been exported, but it cannot be found.";
+            }
+            $vhdFiles += $vhdFullName;
+        }
+        Write-Debug "Exported VHD files: '$vhdFiles'";
+        Write-Output $vhdFiles;
+    }
+
+    function TryGet-HyperVVM([string]$computerName, [string]$hvvmName)
+    {
+        if ([string]::IsNullOrEmpty($computerName))
+        {
+            Write-Warning "Hyper-V computer name cannot be null or empty.";
+            return $false;
+        }
+        if ([string]::IsNullOrEmpty($hvvmName))
+        {
+            Write-Warning "Hyper-V VM name cannot be null or empty.";
+            return $false;
+        }
+
+        try
+        {
+            $allVMs = Get-VM -ComputerName $computerName;
+            $hvvm = $allVMs | where {$_.Name -eq $hvvmName};
+            if ($hvvm -ne $null)
+            {
+                Write-Debug ("Found Hyper-V VM by name '$hvvmName': " + $hvvm);
+                return $true;
+            }
+            else
+            {
+                Write-Warning ("Cannot find Hyper-V VM by name '$hvvmName'...");
+                return $false;
+            }
+        }
+        catch
+        {
+            throw;
+        }
+    }
+
+    if (-not (TryGet-HyperVVM $HyperVServer $HyperVVMName))
+    {
+        throw "Cannot find VM '$HyperVVMName' from server '$HyperVServer'; exit.";
+    }
+
+    $vhdxFolder = Get-HyperVVhdFolder $ExportPath $HyperVVMName;
+    [string[]]$vhdFiles = @();
+    if (Test-Path $vhdxFolder)
+    {
+        if ($Force)
+        {
+            $st = rmdir -Recurse -Force $vhdxFolder;
+        }
+        else
+        {
+            Write-Warning "Export path already exists '$vhdxFolder'; please delete & retry...";
+            return;
+        }
+    }
+    $vhdFiles = Export-DiskFiles $HyperVServer $HyperVVMName $ExportPath;
+
+    # Convert the various VHD/VHDX file formats to fixed VHD files
+    $convertedVhdPath = Join-Path $vhdxFolder "Converted";
+    $st = mkdir $convertedVhdPath -Force;
+    [string[]]$destFiles = @();
+    foreach ($vhdFile in $vhdFiles)
+    {
+        $vhdFileName = Split-Path -Leaf -Path $vhdFile;
+        $destFile = Join-Path $convertedVhdPath $vhdFileName.Replace('.vhdx', '.vhd');
+        Write-Debug "Converting '$($vhdFile)' to '$destFile'...";
```

Comment:
_advisory:_ this should be Write-Verbose

Reply:
ok